common/util/additionalArguments.go

```diff
@@ -0,0 +1,47 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/nutanix-cloud-native/ndb-operator/common"
+)
+
+// Gets allowed additional argument types and indicates whether it is an action argument. Also returns if there is an error or not.
+func GetAllowedAdditionalArgumentsForType(typ string) (map[string]bool, error) {
+	switch typ {
+	case common.DATABASE_TYPE_MSSQL:
+		return map[string]bool{
+			/* Has a default */
+			"sql_user_name":                true,
```

Comment:
i noticed all additional args have a value 'true'. is it fair to say we need a list of valid additional args but since we are using a map, all the values point to 'true'?
can we instead use a list/slice?

Reply:
The true value indicates that the additionalArgument is a actionArgument. Currently, all additionalArgument flows that we support are actionArguments. 

As we add more flows, this might not always be the case. So I thought it might be beneficial to include the 'true' distinction here. What do you think, maybe there's a better alternative?